### PR TITLE
update to mirage 4.5.0

### PR DESCRIPTION
**Pull request by samoht**

### CODE REVIEW

File: mirage/config.ml [hunk 1]
```diff
@@ -155,24 +82,35 @@ let internal_stack =
...
 let mirage_monitoring =
   let ip_key =
     let ip =
       let doc =
-        Key.Arg.info ~doc:"IP of InfluxDB server to transmit metrics to"
+        Arg.info ~doc:"IP of InfluxDB server to transmit metrics to"
           [ "metrics-ip" ]
       in
-      Key.(create "metrics-ip" Arg.(required ip_address doc))
+      Key.create "metrics-ip" (required ip_address doc)
```
Comment (samoht) [inline mirage/config.ml:103]: @hannesm: what's the intended use of this key? This is now failing at configuration time if you don't provide a `--metrics-ip`, which I suspect is not what is expected. Should we turn those keys to runtime keys?

Comment (hannesm) [reply mirage/config.ml:103]: I guess that is the right path to take. The "intended use" is described above, "IP of InfluxDB server to transmit metrics to" -- so yes this should be a runtime key.

Comment (hannesm) [reply mirage/config.ml]: https://github.com/ocaml/opam-repository/pull/25649

**Comment by samoht** [comment]
pfff, all green! I'll cleanup and release the right things now

**Pull request merged**